app/lib/common/widgets/pin/pin_selector_drawer.dart

```diff
@@ -0,0 +1,91 @@
+import 'package:acter/features/events/widgets/skeletons/event_list_skeleton_widget.dart';
+import 'package:acter/features/pins/providers/pins_provider.dart';
+import 'package:acter/features/pins/widgets/pin_list_item_widget.dart';
+import 'package:atlas_icons/atlas_icons.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:logging/logging.dart';
+
+final _log = Logger('a3::common::pin::select_drawer');
+
+const Key selectPinDrawerKey = Key('select-pin-drawer');
+
+Future<String?> selectPinDrawer({
+  required BuildContext context,
+  required String spaceId,
+  Key? key = selectPinDrawerKey,
+  Widget? title,
+}) async {
+  final selected = await showModalBottomSheet(
+    showDragHandle: true,
+    enableDrag: true,
+    context: context,
+    isDismissible: true,
+    builder: (context) => Consumer(
+      builder: (context, ref, child) {
+        final calPinLoader = ref.watch(pinsProvider(spaceId));
+        return calPinLoader.when(
+          data: (pinsList) => Column(
```

Comment:
Ya true, I copied things from previous `selectEventDrawer` code of Event Action Item so that's why forgot to use generic widget code.

Let me use own widget for pin as well event also. Thanks for highlighting it.